### PR TITLE
Replace deprecated Socket.gethostbyname with Addrinfo.getaddrinfo

### PR DESCRIPTION
**Pull request by RyankePars**
Ref: https://apidock.com/ruby/Socket/gethostbyname/class

Works for my use case.

WIP

### CODE REVIEW

File: lib/spf/util.rb [hunk 1]
```diff
@@ -45,7 +45,7 @@ def self.ipv4_mapped_ipv6_address_pattern
   end
 
   def self.hostname
-    return @hostname ||= Socket.gethostbyname(Socket.gethostname).first
+    return @hostname ||= Addrinfo.getaddrinfo(Socket.gethostname, 0).first.getnameinfo.first
```
Comment (jcamenisch) [inline lib/spf/util.rb]: I haven't audited this thoroughly to understand the implications, but I find this preserves existing behavior:
```suggestion
    return @hostname ||= Addrinfo.getaddrinfo(Socket.gethostname, nil, nil, nil, nil, Socket::AI_CANONNAME).first.canonname
```

**Comment by agari-paved-road** [comment]
Closing Pull Request due to no activity in the last 30 days

**Comment by jcamenisch** [comment]
@tweltz @tedaford, I think this would be worth addressing to support newer versions of Ruby, Rails, etc. see https://apidock.com/ruby/Socket/gethostbyname/class

**Comment by agari-paved-road** [comment]
Closing Pull Request due to no activity in the last 30 days

**Comment by agari-paved-road** [comment]
Closing Pull Request due to no activity in the last 30 days

**Review by tedaford — APPROVED** [review]

**Pull request merged**